artifact-manifest-spec.md

```diff
@@ -0,0 +1,77 @@
+# OCI Artifact Manifest Spec (Phase-1 Reference Types)
```

Comment:
This seems like it's been hashed out in the past, but what's to stop us from using the existing image manifest spec and adding *only* the additional fields needed to enable references? In my mind, this would just be `subjectManifest` as an optional field that would allow registries to begin tracking these relationships on push and use that to implement a references or referrers API.

I think that we should at least list out the reasons why we think we need a new manifest type and why we can't use the existing spec. Especially with the conversations going on about another "object manifest" type in the future, I'd really like to keep the number of schemas that registries have to implement to a minimum. Adding fields to the existing spec seems like it *might* be simpler and reduce the overall churn over the next few years.

Reply:
One major issue is that if any object can have a reference (which is a reverse reference for gc) and also a normal link to a manifest, we can no longer guarantee that there are no cycles in the garbage collection graph, and gc becomes intractable.

Reply:
My understanding was that the garbage collector will not be using the `reference` link in the manifest?

Reply:
> gc becomes intractable

I'm not sure if I 100% agree with this, but I do agree that it becomes more complicated. It's hard to really discuss because there are so many different GC policies.

I don't like that this doc prescribes GC behavior, though, as it limits the power of how this kind of relationship could be used.

Reply:
The limitation is simply a minimum scoped set of scenarios we're confident we can support until we have the time to work through all the scenarios in #37.

Reply:
@justincormack is there a restriction preventing an ArtifactManifest from referencing an ArtifactManifest?

Reply:
Nope. See [the example where an SBOM is signed](https://github.com/SteveLasker/artifacts/blob/oci-artifact-manifest/artifact-manifest.md#notary-v2-signatures-and-sbom-persistance). The SBOM is an `artifact reference type`, and the signature is also an `artifact referenceType`

![image](https://user-images.githubusercontent.com/7647382/125677271-ee7f9df1-2006-44c7-80da-4c4576c05a3d.png)